sql/src/main/java/io/crate/operation/collect/LuceneDocCollector.java

```diff
@@ -290,7 +291,7 @@ private void innerCollect() {
     }
 
     private void finishCollect() {
-        if (!paused.get()) {
+        if (finished) {
```

Comment:
when replacing paused boolean with finished one here, afaik finished must be set at all catch cases at innerCollect where paused was set to false...